test/e2e/tests/infra/infra.go

```diff
@@ -0,0 +1,336 @@
+package infra
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"time"
+
+	mgmtgraphrbac "github.com/Azure/azure-sdk-for-go/services/graphrbac/1.6/graphrbac"
+	mgmtfeatures "github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2019-07-01/features"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/Azure/go-autorest/autorest/date"
+	"github.com/Azure/go-autorest/autorest/to"
+	uuid "github.com/satori/go.uuid"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/client/services/redhatopenshift/mgmt/2020-04-30/redhatopenshift"
+	"github.com/Azure/ARO-RP/pkg/util/arm"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/graphrbac"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/features"
+	mgmtaro "github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/redhatopenshift"
+	"github.com/Azure/ARO-RP/pkg/util/azureerrors"
+)
+
+const FP_CLIENT_ID = "1516efbc-0d48-4ade-a68e-a2872137fd79"
+
+type Infrastructure struct {
+	conf         Config
+	deployments  features.DeploymentsClient
+	rgClientSet  features.ResourceGroupsClient
+	appClientSet graphrbac.ApplicationsClient
+	spClientSet  graphrbac.ServicePrincipalClient
+	aroClientSet mgmtaro.OpenShiftClustersClient
+
+	appOID string
+}
+
+const letterBytes = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+
+func randStringBytes(n int) string {
+	b := make([]byte, n)
+	for i := range b {
+		b[i] = letterBytes[rand.Intn(len(letterBytes))]
+	}
+	return string(b)
+}
```

Comment:
Why not something like `strconv.FormatInt(time.Now().Unix(), 10)` where we get untixtime stamp. We can use it later for cleaning if needed too.

Reply:
This did not occur to me, it is good idea. Thanks